src/components/Filters/SchemaSieve.ts

```diff
@@ -1,157 +1,516 @@
-	FilterModel,
-	FilterRule,
-	PineTypeModule,
-	Schema,
-	SchemaEntry,
-	SchemaSieveClass,
-} from 'rendition';
-import filterTests from '../PineTypes';
-
-type PineTypeModuleStruct = { [key: string]: PineTypeModule };
-
-export const SIMPLE_SEARCH_NAME = 'Full text search';
-
-class SchemaSieve implements SchemaSieveClass {
-	public SIMPLE_SEARCH_NAME = SIMPLE_SEARCH_NAME;
-	public tests: PineTypeModuleStruct;
-
-	constructor(tests: PineTypeModuleStruct = {}) {
-		this.tests = assign(cloneDeep(filterTests), tests);
-	}
-
-	test(item: any, input: FilterRule | FilterRule[]) {
-		return isArray(input)
-			? every(input, i => this.baseTest(item, i))
-			: this.baseTest(item, input);
-	}
-
-	baseTest(item: any, input: FilterRule): boolean {
-		// If this is a compound rule, evaluate the rules together using OR logic
-		if (input.extra && input.extra.or) {
-			return (
-				this.baseTest(item, omit(input, 'extra') as FilterRule) ||
-				some(input.extra.or, rule => this.baseTest(item, rule as FilterRule))
-			);
-		}
+import { FilterSignature, FiltersView, v3, ViewScope } from 'rendition';
+import * as utils from '../../utils';
+import DataTypes from '../DataTypes';
 
-		const { type, name } = input;
-
-		// A simple search is not strictly a "type" and searches on all fields,
-		// so we handle that seperately here
-		if (name === this.SIMPLE_SEARCH_NAME) {
-			const { value } = input;
-			return some(
-				item,
-				target =>
-					target && includes(String(target).toLowerCase(), value.toLowerCase()),
-			);
-		}
+const ajv = new Ajv();
+ajvKeywords(ajv);
+ajv.addMetaSchema(metaSchema6);
+
+const FULL_TEXT_SLUG = 'full_text_search';
+
+export const filter = (
+	filters: JSONSchema6 | JSONSchema6[],
+	collection: any[],
+) => {
+	// Remove all schemas that may have been compiled already
+	ajv.removeSchema(/^.*$/);
+
+	const validators = isArray(filters)
+		? filters.map(s => ajv.compile(s))
+		: [ajv.compile(filters)];
+
+	if (isArray(collection)) {
+		return collection.filter(m => every(validators, v => v(m)));
+	}
 
-		if (type in filterTests) {
-			const { operator, value } = input;
-			const target = item[name];
+	return pickBy(collection, m => every(validators, v => v(m)));
+};
 
-			if (operator! in filterTests[type].rules) {
-				const result = filterTests[type].rules[operator!];
-				return isPlainObject(result)
-					? result.test(target, value)
-					: result(target, value);
-			}
+export const getDataModel = (schema?: JSONSchema6) => {
+	if (!schema) {
+		return null;
+	}
+
+	const { format, type } = schema;
+
+	if (schema.enum) {
+		return DataTypes.enum;
+	}
+	if (type === 'string') {
+		if (format === 'date-time') {
+			return DataTypes.date_time;
 		}
+		return DataTypes.string;
+	}
+	if (type === 'object') {
+		return DataTypes.object;
+	}
+	if (type === 'boolean') {
+		return DataTypes.boolean;
+	}
+	if (type === 'number') {
+		return DataTypes.number;
+	}
+
+	return null;
+};
+
+export const createFullTextSearchFilter = (
```

Comment:
Would it make sense to move the FullTextSearch related methods to a different file?
This is yet an other implementation of a filter type, so it might not have to be next to the sieve methods, even though re-exporting is required.